org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/theme/Theme.java

```diff
@@ -226,32 +222,39 @@ public static Theme resolveParsedThemeRules(final List<ParsedThemeRule> parsedTh
 		return new Theme(colorMap, defaults, root);
 	}
 
-	public Theme(final ColorMap colorMap, final ThemeTrieElementRule defaults, final ThemeTrieElement root) {
-		this.colorMap = colorMap;
-		this.root = root;
-		this.defaults = defaults;
-	}
+	private static boolean isValidHexColor(final String hex) {
+		if (hex.isEmpty()) {
+			return false;
+		}
 
-	public Set<String> getColorMap() {
-		return this.colorMap.getColorMap();
+		if (RRGGBB.matcher(hex).matches()) {
+			// #rrggbb
+			return true;
+		}
+
+		if (RRGGBBAA.matcher(hex).matches()) {
+			// #rrggbbaa
+			return true;
+		}
+
+		if (RGB.matcher(hex).matches()) {
+			// #rgb
+			return true;
+		}
+
+		if (RGBA.matcher(hex).matches()) {
+			// #rgba
+			return true;
+		}
```

Comment:
```suggestion
		return RRGGBB.matcher(hex).matches() || RRGGBBAA.matcher(hex).matches() || RGB.matcher(hex).matches() || RGBA.matcher(hex).matches();
```

Reply:
The current code base tries to stay close to the upstream typescript project. so if bugs are solved there they can be easily ported to tm4e by visual comparison of the code bases. The equivalent code in vscode is: https://github.com/microsoft/vscode-textmate/blob/9157c7f869219dbaf9a5a5607f099c00fe694a29/src/utils.ts#L132-L154